Sources/TableViewDriver.swift

```diff
@@ -375,7 +375,7 @@ extension TableViewDriver: UITableViewDelegate {
 
     /// :nodoc:
     public func tableView(_ tableView: UITableView, willSelectRowAt indexPath: IndexPath) -> IndexPath? {
-        if let shouldSelect = self.tableViewModel?[ifExists: indexPath]?.shouldSelect, !shouldSelect {
+        if let shouldSelect = self.tableViewModel?[ifExists: indexPath]?.shouldSelect(at: indexPath), !shouldSelect {
```

Comment:
I don't like how this reads -- I know we access the model in a similar way in other functions but this would be better:
```
guard let viewModel = self.tableViewModel?[ifExists: indexPath] else {
    return indexPath
}
return viewModel.shouldSelect(at: indexPath) ? indexPath : nil
```